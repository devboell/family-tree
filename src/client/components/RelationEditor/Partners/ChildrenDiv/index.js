import styled from 'styled-components'

const ChildrenDiv = styled.div`
  margin-left: 40px;
  > p {
    color: gray;
  }
`

export default ChildrenDiv
