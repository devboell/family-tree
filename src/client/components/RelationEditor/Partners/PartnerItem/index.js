import styled from 'styled-components'

const PartnerItem = styled.div`
  display: flex;
  margin-bottom: 3px;
  > p {
    padding-top: 2px;
  }
`

export default PartnerItem
