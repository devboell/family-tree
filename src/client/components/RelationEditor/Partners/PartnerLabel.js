import styled from 'styled-components'

const PartnerLabel = styled.p`
  font-weight: bold;
  margin-bottom: 10px;
`

export default PartnerLabel
