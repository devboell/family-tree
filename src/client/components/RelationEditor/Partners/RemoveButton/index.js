import styled from 'styled-components'

const RemoveButton = styled.button`
  padding: 5px;
  margin-right: 10px;
  cursor: pointer;
`

export default RemoveButton
