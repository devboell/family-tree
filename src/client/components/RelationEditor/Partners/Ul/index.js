import styled from 'styled-components'


const Ul = styled.ul`
  list-style: none;
`

export default Ul
