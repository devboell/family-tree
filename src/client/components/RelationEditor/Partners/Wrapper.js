import styled from 'styled-components'

const Wrapper = styled.div`
  flex-basis: 50%;
  margin-left: 10px;
`

export default Wrapper
