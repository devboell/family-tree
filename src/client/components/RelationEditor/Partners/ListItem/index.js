import React from 'react'
import { PropTypes } from 'prop-types'

import PartnerItem from './PartnerItem'
import RemoveButton from './RemoveButton'
import ChildrenDiv from './ChildrenDiv'

const ListItem = ({ input: { value }, onRemove }) =>
  <li>
    <PartnerItem>
      <RemoveButton type="button" onClick={onRemove}>X</RemoveButton>
      <p>{value.partner.name}</p>
    </PartnerItem>
    <ChildrenDiv>
      {value.children.map(child => (
        <p
          key={`child-${child.name}`}
        >
          {child.name}
        </p>
      ))}
    </ChildrenDiv>
  </li>

ListItem.propTypes = {
  input: PropTypes.shape({}).isRequired,
  onRemove: PropTypes.func.isRequired,
}

export default ListItem
