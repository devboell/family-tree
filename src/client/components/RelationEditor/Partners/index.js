import React from 'react'
import { PropTypes } from 'prop-types'
import { Field } from 'redux-form'

import ListItem from './ListItem'
import Wrapper from './Wrapper'
import PartnerLabel from './PartnerLabel'
import Ul from './Ul'


const Partners = ({ fields }) =>
  <Wrapper>
    <PartnerLabel>Partners:</PartnerLabel>
    <Ul>
      {fields.map((partner, i) => (
        <Field
          key={`partner-${partner}`}
          name={partner}
          component={ListItem}
          onRemove={() => { fields.remove(i) }}
        />
      ))}
    </Ul>
  </Wrapper>


Partners.propTypes = {
  fields: PropTypes.shape({}).isRequired,
}


export default Partners
