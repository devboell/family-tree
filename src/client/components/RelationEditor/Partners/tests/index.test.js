import React from 'react'
import Partners from '../index'


// TODO: find way to properly test redux-form
const partners = [
  'John',
  'Mary',
]

const props = {
  fields: {
    remove: jest.fn(),
    name: 'partners',
    map: iter => partners.map(iter),
  },
}

it('AvailablePartners, renders without crashing', () => {
  expect(shallow(<Partners {...props} />)).toMatchSnapshot()
})
