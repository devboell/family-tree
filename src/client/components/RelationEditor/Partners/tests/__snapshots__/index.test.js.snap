// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvailablePartners, renders without crashing 1`] = `
<styled.div>
  <styled.p>
    Partners:
  </styled.p>
  <styled.ul>
    <Field
      component={[Function]}
      name="John"
      onRemove={[Function]}
    />
    <Field
      component={[Function]}
      name="Mary"
      onRemove={[Function]}
    />
  </styled.ul>
</styled.div>
`;
